includes/demo.php

```diff
@@ -84,173 +84,129 @@ function wp_notify_add_admin_options_page() {
-			<th scope="col" id="sms" class="manage-column column-format column-sms desc"><span>SMS</span></th>
-			<th scope="col" id="app" class="manage-column column-format column-app desc"><span>App</span></th>
-		</tr>
-		</tfoot>
-	</table>
-
 	<?php
+	/**
+	 * List_table that shows the settings
+	 */
+	class List_Table extends WP_List_Table {
+		/**
+		 * Prepare the items for the table to process
+		 */
+		public function prepare_items() {
+			$columns  = $this->get_columns();
+			$hidden   = $this->get_hidden_columns();
+			$sortable = $this->get_sortable_columns();
+
+			$data = $this->table_data();
+
+			$per_page     = 10;
+			$current_page = $this->get_pagenum();
+			$total_items  = count( $data );
+
+			$this->set_pagination_args(
+				array(
+					'total_items' => $total_items,
+					'per_page'    => $per_page,
+				)
+			);
+
+			$data = array_slice( $data, ( ( $current_page - 1 ) * $per_page ), $per_page );
+
+			$this->_column_headers = array( $columns, $hidden, $sortable );
+			$this->items           = $data;
+		}
+
+		/**
+		 * Override the parent columns method. Defines the columns to use in your listing table
+		 *
+		 * @return Array
+		 */
+		public function get_columns(): array {
+			return array(
+				'cb'     => '<input type="checkbox" />',
+				'source' => __( 'Notification Source' ),
+				'show'   => __( 'Show in:' ),
+				'admin'  => __( 'Admin' ),
+				'email'  => __( 'Email' ),
+				'sms'    => __( 'Sms' ),
+				'app'    => __( 'App' ),
+			);
+		}
+
+		public function get_hidden_columns(): array {
+			return array();
+		}
+
+		public function get_sortable_columns(): array {
+			return array( 'source' => array( 'source', false ) );
+		}
+
+		private function table_data(): array {
+			$data = array();
+
+			$indexes = array(
+				'WordPress',
+				'User activity (published, edited, ect)',
+				'Comments',
+				'Site health',
+			);
+
+			$item_count = count( $indexes );
+
+			for ( $i = 0; $i < $item_count; $i++ ) {
+				$label      = $indexes[ $i ];
+				$data[ $i ] = array(
+					'cb'     => $i,
+					'source' => $label,
+					'show'   => '',
+					'admin'  => '<input type="checkbox" id="cb-admin-' . $i . '" />',
+					'email'  => '<input type="checkbox" id="cb-email-' . $i . '" />',
+					'sms'    => '<input type="checkbox" id="cb-sms-' . $i . '" />',
+					'app'    => '<input type="checkbox" id="cb-app-' . $i . '" />',
+				);
+			}
+			return $data;
+		}
+
+		public function column_default( $item, $column_name ) {
+			switch ( $column_name ) {
+				case 'cb':
+				case 'source':
+				case 'show':
+				case 'admin':
+				case 'email':
+				case 'sms':
+				case 'app':
+					return $item[ $column_name ];
+
+				default:
+					return $item;
+			}
+		}
+	}
+
+	/**
+	 * The settings table
+	 *
+	 * @return void
+	 */
+	function init_table() {
+		/** @var $list_table $list_table */
+		$list_table = new list_table();
+		$list_table->prepare_items();
+		$list_table->display();
```

Comment:
```suggestion
		$list_table->display();
		echo '<input type="submit" id="doaction" class="button action primary" value="apply">';
```

will add the missing confirmation button

Reply:
If you're going to add this, could you make the `apply` uppercase and translatable?

Reply:
```suggestion
		$list_table->display();
		printf( '<input type="submit" id="doaction" class="button action primary" value="%s">', __( 'Apply' ) );
```

right... I would say it is now OK, only the text domain is missing but at least on this page it has never been used yet